### PR TITLE
feat(authors backfill): [BACK-1460+BACK-1461] Create a lambda for Authors Backfill

### PR DESCRIPTION
**Pull request by nina-py**
## Goal

Set up infrastructure for the Author Backfill lambda that should include, among other things, an SQS queue a custom script should push its messages to (see details and a related PR in the ticket).

Note that tests are commented out until the lambda is actually set up to be used for its purpose in BACK-1461.

## References

JIRA ticket:
* https://getpocket.atlassian.net/browse/BACK-1460
* https://getpocket.atlassian.net/browse/BACK-1461

### CODE REVIEW

**Comment by nina-py** [comment]
Thank you @jpetto! Now I just have to fix the tests :)

File: .aws/src/backfillLambda.ts [hunk 1]
```diff
@@ -123,8 +123,8 @@ export class BackfillLambda extends Resource {
       threshold: 4000,
       // rather than the average, we want to track how many messages in total end up in the DLQ
       statistic: 'Sum',
-      alarmActions: [this.pagerDuty.snsNonCriticalAlarmTopic.arn],
-      okActions: [this.pagerDuty.snsNonCriticalAlarmTopic.arn],
+      alarmActions: [this.pagerDuty?.snsNonCriticalAlarmTopic.arn],
+      okActions: [this.pagerDuty?.snsNonCriticalAlarmTopic.arn],
```
Comment (nina-py) [inline .aws/src/backfillLambda.ts:127]: This change was the last change needed to be able to deploy to Dev - apparently, in the backfill lambda most of this code is copied from alarms were not set up at the time testing on Dev was in progress, they were set up later on when test runs were only done on Prod, so this error has not come up before.

File: curation-authors-backfill/externalCaller/curatedCorpusApiCaller.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { getCurationToolsDataSyncPrivateKey } from '../secretManager';
+
+/**
+ * Call the importApprovedCorpusItem mutation
+ * @param data input variables for the mutation. Of the type CorpusInput
+ * @returns json response of type ImportApprovedCorpusItemPayload
+ */
+export async function importApprovedCorpusItem(data: CorpusInput) {
+  const mutation = `
+  mutation importApprovedItem($data: ImportApprovedCorpusItemInput!) {
```
Comment (sri-kirsh) [inline curation-authors-backfill/externalCaller/curatedCorpusApiCaller.ts:14]: did we add the `authors` field as a part of mutation input?

Comment (sri-kirsh) [reply curation-authors-backfill/externalCaller/curatedCorpusApiCaller.ts:14]: looks like we need to use `UpdateApprovedCorpusItemAuthors` mutation for updating authors https://github.com/Pocket/curated-corpus-api/blob/main/schema-admin.graphql#L724

File: curation-authors-backfill/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    // mock the secrets manager call
+    sinon
+      .stub(SecretManager, 'getCurationToolsDataSyncPrivateKey')
+      .resolves('test-secret');
+
+    // mock the generate jwt function
+    sinon.stub(Jwt, 'generateJwt').returns('test-jwt');
+    // sinon
+    //   .stub(curatedItemIdMapper, 'insertCuratedItem')
+    //   .callsFake(stubInsertCuratedItem);
```
Comment (jpetto) [inline curation-authors-backfill/index.spec.ts:32]: can we drop these lines?

File: curation-authors-backfill/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+import * as Jwt from './jwt';
+import sinon from 'sinon';
+import nock from 'nock';
+import config from './config';
+import { SQSEvent } from 'aws-lambda';
+import * as ProspectApi from './externalCaller/prospectApiCaller';
+
+// async function stubInsertCuratedItem() {
+//   return;
+// }
```
Comment (jpetto) [inline curation-authors-backfill/index.spec.ts:12]: can we drop this?

File: curation-authors-backfill/index.spec.ts [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+              approvedItem: {
+                externalId: 'random-approvedItem-guid',
+              },
+              scheduledItem: {
+                externalId: 'random-scheduledItem-guid',
+                scheduledSurfaceGuid: 'new_tab_en_us',
+              },
+            },
+          },
+        })
```
Comment (jpetto) [inline curation-authors-backfill/index.spec.ts:66]: this should be modified to our authors backfill scenario, yeah?

**Review by jpetto — CHANGES_REQUESTED** [review]
i think this needs just a couple tweaks to satisfy the acceptance criteria of the ticket:

1. log the data we intend to send to the mutation
2. clear out/stub the `index.spec.ts` file, as the one in this PR seems to be mostly (all?) copy pasta

i may be able to add this commit today.

File: curation-authors-backfill/index.spec.ts (hunk 3 above)
Comment (nina-py) [reply curation-authors-backfill/index.spec.ts:32]: Done!

File: curation-authors-backfill/index.spec.ts (hunk 5 above)
Comment (nina-py) [reply curation-authors-backfill/index.spec.ts:66]: I've updated them all and there's no need to be skipping any of those tests any more.

**Review by jpetto — APPROVED** [review]
time to merge and iterate! :shipit:

**Pull request merged**